src/MyConnection.h

```diff
@@ -6,6 +6,10 @@
 #include <boost/noncopyable.hpp>
 #include <boost/shared_ptr.hpp>
 
+const char* as_api_string(const Rcpp::Nullable<std::string>& cpp_string) {
```

Comment:
I think this could work if we use Rcpp::Nullable<CharacterVector> as type and access the underlying R string via CHAR(user[0]) as suggested by Hadley.